cmd/deploy/deploy.go

```diff
@@ -286,18 +287,28 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		return oktetoErrors.ErrDeployCantDeploySvcsIfNotCompose
 	}
 
+	// We need to create a client that doesn't go through the proxy to create
+	// the configmap without the deployedByLabel
+	c, _, err := dc.K8sClientProvider.Provide(okteto.Context().Cfg)
+	if err != nil {
+		return err
+	}
+
 	cwd, err := os.Getwd()
 	if err != nil {
 		return fmt.Errorf("failed to get the current working directory: %w", err)
 	}
 
-	// We need to create a client that doesn't go through the proxy to create
-	// the configmap without the deployedByLabel
-	c, _, err := dc.K8sClientProvider.Provide(okteto.Context().Cfg)
+	topLevelGitDir, err := repository.FindTopLevelGitDir(cwd, afero.NewOsFs())
 	if err != nil {
-		return err
+		oktetoLog.Warning("Some git env vars might not be available: %s", err.Error())
```

Comment:
This is a warning in case the user did not notice, but we don't stop the execution to allow users run `okteto` in directories that are not git repos.

The warning looks like this:
![Screenshot 2023-12-05 at 10 21 12](https://github.com/okteto/okteto/assets/2318450/854f3573-807f-409b-9285-b678efd67c3a)

Reply:
Agree but I would use the info level, since it´s a very common use case that should not be debugged with such a low log level.

Reply:
I am wondering how frequently it's a common use case against how many times this could occur by mistake? I suspect that most people use okteto with git repos and that's why I went for warning, because it would be very helpful to let the user know about a mistake, however I don't have the full visibility on usage so I would appreciate input from @rlamana

Reply:
I'm trying to understand this particular case, will this be shown when no `.git` folder is found? What use cases cover this situation?

Reply:
This warning appears when the `.git` cannot be found (up until `/`).

Imagine I work without any version control, and I want to use `okteto`, I would get this warning. The idea is to warn users that do use git, but for some reasons are running `okteto deploy` and the `.git` cannot be found.

Reply:
Updated in 0f5f092e

![Screenshot 2023-12-07 at 14 57 54](https://github.com/okteto/okteto/assets/2318450/f232a498-d9b6-4afa-af1c-8cee8b9875b4)